app/gui2/src/stores/suggestionDatabase/index.ts

```diff
@@ -56,19 +56,19 @@ class Synchronizer {
   queue: AsyncQueue<{ currentVersion: number }>
 
   constructor(
+    projectStore: ProjectStore,
     public entries: SuggestionDb,
     public groups: Ref<Group[]>,
   ) {
-    const projectStore = useProjectStore()
     const lsRpc = projectStore.lsRpcConnection
     const initState = exponentialBackoff(() =>
       lsRpc.acquireCapability('search/receivesSuggestionsDatabaseUpdates', {}),
     ).then((capability) => {
       if (!capability.ok) {
         capability.error.log('Will not receive database updates')
       }
-      this.setupUpdateHandler(lsRpc)
-      this.loadGroups(lsRpc, projectStore.firstExecution)
+      this.#setupUpdateHandler(lsRpc)
+      this.#loadGroups(lsRpc, projectStore.firstExecution)
```

Comment:
Why do we change it? Isn't TS compile-time checking sufficient?

Reply:
In this case the use of `private` caused weird issues with typechecking after the types went through `proxyRefs`. I just decided to use native private fields to get around that, as those have no such issues.